### PR TITLE
statistics: upgrade stats timeout checkpoint after it timeouts

### LINKED ISSUE #52425: forget to upgrade the time after the load stats timeout
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

image

after task timeout, we should upgrade the timeout checkpoint.

if we retry it immediately, it will timeout as soon as possible.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #52425

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: pkg/sessionctx/variable/session.go [hunk 1]
```diff
@@ -1307,7 +1307,7 @@ type SessionVars struct {
 	ReadConsistency ReadConsistencyLevel
 
 	// StatsLoadSyncWait indicates how long to wait for stats load before timeout.
-	StatsLoadSyncWait int64
+	StatsLoadSyncWait atomic.Int64
```
Comment (hawkingrei) [inline pkg/sessionctx/variable/session.go:1310]: Update it to resolve the data race problem.

**Review by AilinKid — APPROVED** [review]

File: pkg/statistics/handle/syncload/stats_syncload.go [hunk 2]
```diff
@@ -396,7 +397,7 @@ func (*statsSyncLoad) readStatsForOneItem(sctx sessionctx.Context, item model.Ta
 }
 
 // drainColTask will hang until a column task can return, and either task or error will be returned.
-func (s *statsSyncLoad) drainColTask(exit chan struct{}) (*statstypes.NeededItemTask, error) {
+func (s *statsSyncLoad) drainColTask(sctx sessionctx.Context, exit chan struct{}) (*statstypes.NeededItemTask, error) {
```
Comment (AilinKid) [inline pkg/statistics/handle/syncload/stats_syncload.go:400]: pass the limited scope is appreciated

**Pull request merged**